react-effects/my-app/src/List.jsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+  useEffect(() => {
+    readItems()
+      .then((list) => {
+        setItems(list);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setError(error);
+        setIsLoading(false);
```

Comment:
Instead of having to write `setIsLoading(false`) twice, we can use a finally!

```
  useEffect(() => {
    readItems()
    .then((list) => setItems(list))
    .catch((error) => setError(error))
    .finally(()=> setIsLoading(false));
  }, []);
```